plugins/outputs/timestream/timestream.go

```diff
@@ -0,0 +1,616 @@
+			return fmt.Errorf("in '%s' mapping mode, SingleTableDimensionNameForTelegrafMeasurementName key is required",
+				MappingModeSingleTable)
+		}
+	}
+
+	if t.MappingMode == MappingModeMultiTable {
+		if t.SingleTableName != "" {
+			return fmt.Errorf("in '%s' mapping mode, do not specify SingleTableName key", MappingModeMultiTable)
+		}
+
+		if t.SingleTableDimensionNameForTelegrafMeasurementName != "" {
+			return fmt.Errorf("in '%s' mapping mode, do not specify SingleTableDimensionNameForTelegrafMeasurementName key", MappingModeMultiTable)
+		}
+	}
+
+	if t.CreateTableIfNotExists {
+		if t.CreateTableMagneticStoreRetentionPeriodInDays < 1 {
+			return fmt.Errorf("if Telegraf should create tables, CreateTableMagneticStoreRetentionPeriodInDays key should have a value greater than 0")
+		}
+
+		if t.CreateTableMemoryStoreRetentionPeriodInHours < 1 {
+			return fmt.Errorf("if Telegraf should create tables, CreateTableMemoryStoreRetentionPeriodInHours key should have a value greater than 0")
+		}
+	}
+
+	log.Printf("I! Timestream: Constructing Timestream client for '%s' mode", t.MappingMode)
+
+	credentialConfig := &internalaws.CredentialConfig{
+		Region:      t.Region,
+		AccessKey:   t.AccessKey,
+		SecretKey:   t.SecretKey,
+		RoleARN:     t.RoleARN,
+		Profile:     t.Profile,
+		Filename:    t.Filename,
+		Token:       t.Token,
+		EndpointURL: t.EndpointURL,
+	}
+	svc := WriteFactory(credentialConfig)
+
+	if t.DescribeDatabaseOnStart {
+		log.Printf("I! Timestream: Describing database '%s' in region '%s'", t.DatabaseName, t.Region)
+
+		describeDatabaseInput := &timestreamwrite.DescribeDatabaseInput{
+			DatabaseName: aws.String(t.DatabaseName),
+		}
+		describeDatabaseOutput, err := svc.DescribeDatabase(describeDatabaseInput)
+		if err != nil {
+			log.Printf("E! Timestream: Couldn't describe database '%s'. Check error, fix permissions, connectivity, create database.", t.DatabaseName)
+			return err
+		}
+		log.Printf("I! Timestream: Describe database '%s' returned: '%s'.", t.DatabaseName, describeDatabaseOutput)
+	}
+
+	t.svc = svc
+	return nil
+}
+
+func (t *Timestream) Close() error {
+	return nil
+}
+
+func (t *Timestream) SampleConfig() string {
+	return sampleConfig
+}
+
+func (t *Timestream) Description() string {
+	return "Configuration for Amazon Timestream output."
+}
+
+func init() {
+	outputs.Add("timestream", func() telegraf.Output {
+		return &Timestream{}
+	})
+}
+
+func (t *Timestream) Write(metrics []telegraf.Metric) error {
+	writeRecordsInputs := t.TransformMetrics(metrics)
+	for _, writeRecordsInput := range writeRecordsInputs {
+		if err := t.writeToTimestream(writeRecordsInput, true); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (t *Timestream) writeToTimestream(writeRecordsInput *timestreamwrite.WriteRecordsInput, resourceNotFoundRetry bool) error {
+	if t.Debug {
+		log.Printf("I! Timestream: Writing to Timestream: '%v' with ResourceNotFoundRetry: '%t'", writeRecordsInput, resourceNotFoundRetry)
```

Comment:
instead of adding a Debug parameter and using log.Printf, you can just use the `Log telegraf.Logger` you already added to the struct, which has a Debug method, and respects the global debug flags. For tests, set the logger to `testutil.Logger{}`. Then you can also remove the LogLevel and plugin name from the log lines and they will conform to the Telegraf standards.